### PR TITLE
add missing cyclic relations

### PR DESCRIPTION
**Pull request by sjib**
should solve https://github.com/teksi/wastewater/issues/107

### CODE REVIEW

**Comment by sjib** [comment]
@3nids Are these the right options for a cyclic relation?
` ON UPDATE CASCADE ON DELETE SET NULL DEFERRABLE INITIALLY DEFERRED`

**Comment by ponceta** [comment]
@ponceta check the defferable setting with @3nids before merging this.

**Review by 3nids — APPROVED** [review]

**Comment by ponceta** [comment]
The differences between DEFERRABLE INITIALLY DEFERRED, DEFERRABLE INITIALLY IMMEDIATE and NOT DEFERRABLE are as follows:

* DEFERRABLE INITIALLY DEFERRED means that the constraint is deferred until the end of the transaction, unless the SET CONSTRAINTS command is used to make it immediate.
* DEFERRABLE INITIALLY IMMEDIATE means that the constraint is checked at the end of each instruction, unless the SET CONSTRAINTS command is used to make it deferred.
* NOT DEFERRABLE means that the constraint is always checked at the end of each instruction, and that the SET CONSTRAINTS command cannot be used to modify it.

In summary, DEFERRABLE and NOT DEFERRABLE indicate whether the constraint can be deferred or not, and INITIALLY DEFERRED and INITIALLY IMMEDIATE indicate the default checking mode for the constraint.

@sjib, @3nids, @cymed, @domi4484  for TEKSI would you be OK to use the DEFERRABLE INITIALLY DEFERRED for every relation connecting objects with each other in the database? (at least where it makes sense, so we can set it once and for all for the database)

**Comment by sjib** [comment]
> @ponceta Good for me - please merge this and #156 I will create a version with DEFERRABLE INITIALLY DEFERRED in a new pull request based on the updated master.

@ponceta Here the additional pull request: https://github.com/teksi/wastewater/pull/178
